classes/task/emit_task.php

```diff
@@ -84,6 +85,36 @@ private function record_successful_events($events) {
         mtrace(count($this->get_successful_events($events)) . " " . get_string('successful_events', 'logstore_xapi'));
     }
 
+    /**
+     * Take event data from the current task
+     * and add to the sent log if it doesn't exist already.
+     *
+     * @param array $event raw event data
+     */
+    private function add_event_to_sent_log($event) {
```

Comment:
Do you have any stats on the performance of this function?
From speaking with Ryan, it looks as though 1500 is the optimal batch size for sending a request to the LRS.

Reply:
On the docker-moodle instance on my Mac with a 2.25GHz CPU with 16Gb RAM and SSD it took the following times to add 1500 rows on 3 separate tests:
1.8529641628265 seconds
1.689178943634 seconds
1.920863866806 seconds